pkg/backup/backup_compaction_test.go

```diff
@@ -6,67 +6,94 @@
 package backup
 
 import (
+	"context"
 	"fmt"
 	"net/url"
 	"strings"
 	"testing"
 
+	"github.com/cockroachdb/cockroach/pkg/backup/backupinfo"
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/stretchr/testify/require"
 )
 
 func TestBackupCompaction(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
+	ctx := context.Background()
 	tempDir, tempDirCleanup := testutils.TempDir(t)
 	defer tempDirCleanup()
+	st := cluster.MakeTestingClusterSettings()
+	backupinfo.WriteMetadataWithExternalSSTsEnabled.Override(ctx, &st.SV, true)
 	_, db, cleanupDB := backupRestoreTestSetupEmpty(
-		t, singleNode, tempDir, InitManualReplication, base.TestClusterArgs{},
+		t, singleNode, tempDir, InitManualReplication, base.TestClusterArgs{
+			ServerArgs: base.TestServerArgs{
+				Settings: st,
+			},
+		},
 	)
 	defer cleanupDB()
-	fullBackupStmt := "BACKUP INTO 'nodelocal://1/backup'"
 
+	compactionBuiltin := "SELECT crdb_internal.backup_compaction(ARRAY['nodelocal://1/backup'], 'LATEST', ''::BYTES, '%s'::DECIMAL, '%s'::DECIMAL)"
 	t.Run("compaction creates a new backup", func(t *testing.T) {
-		numExplicitBackups := 3
-		db.Exec(t, fullBackupStmt)
+		start := getTime()
+		db.Exec(
+			t,
+			fmt.Sprintf("BACKUP INTO 'nodelocal://1/backup' AS OF SYSTEM TIME '%s'", start),
```

Comment:
nit: create `fullBackupAostCmd ="BACKUP INTO 'nodelocal://1/backup' AS OF SYSTEM TIME '%s'"` as you use it in a bunch of subtests. You could also create `incBackupCmd` and `incBackupAOSTCmd` vars to use throughout. 

(string vars  feel easier to read rather than the function i told you inline in an earlier pr, if that makes sense)

Reply:
Yeah that sounds good to me, it does make it easier to see at a glance what type of backup I'm running